exercises/book_changeset.livemd

```diff
@@ -117,9 +117,52 @@ You should rely on Ecto Changesets and your own custom validation to validate th
 
 ```elixir
 defmodule Book do
+  @mail_regex ~r/^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$/
+
+  @types %{
+    title: :string,
+    content: :string,
+    published_on: :date,
+    books_sold: :integer,
+    category: :string,
+    publisher_email: :string,
+    author: :string,
+    has_license: :boolean,
+    price: :integer
+  }
+
+  @keys Map.keys(@types)
+
+  defstruct @keys
+
+  def changeset(%Book{} = book, params \\ %{}) do
+    {book, @types}
+    |> Ecto.Changeset.cast(params, Map.keys(@types))
+    |> Ecto.Changeset.validate_required([:title, :content])
+    |> Ecto.Changeset.validate_inclusion(:category, ["action", "fiction", "mystery"])
+    |> Ecto.Changeset.validate_number(:books_sold, greater_than_or_equal_to: 0)
+    |> Ecto.Changeset.validate_format(:publisher_email, @mail_regex)
```

Comment:
This is better than the example solution! I should have used validate_format but instead wrote a custom validation.

Would you be interested in making a PR to make this the example solution?